src/Fixer/ArrayNotation/MultilineLongArrayFixer.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ArrayNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Philippe Bouttereux <philippe.bouttereux@gmail.com>
+ */
+final class MultilineLongArrayFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    private int $maxArrayLen = 0;
+
+    public function configure(array $configuration): void
+    {
+        parent::configure($configuration);
+
+        $this->maxArrayLen = $this->configuration['max_length'] ?? 0;
+    }
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'A single-line array should be broken into multiple lines if it exceeds configured limit. Arrays that contain comments should be left unchanged.',
+            [
+                new CodeSample("<?php\n\$array = ['a very very long element','another very long element'];\n"),
+                new CodeSample("<?php\n\$array = ['a very very long element','another very long element'];\n", ['max_length' => 10]),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before ArrayIndentationFixer, NoTrailingCommaInSinglelineFixer, TrailingCommaInMultilineFixer, WhitespaceAfterCommaInArrayFixer.
+     */
+    public function getPriority(): int
+    {
+        return 50;
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isAnyTokenKindsFound([T_ARRAY, CT::T_ARRAY_SQUARE_BRACE_OPEN]);
```

Comment:
I believe fixer should react also to `CT::T_DESTRUCTURING_SQUARE_BRACE_OPEN` and handle code like:

```php
// Before
[$foo, $bar, $baz] = $arr;

// After
[
    $foo,
    $bar,
    $baz
] = $arr;
```

Reply:
Makes sense. I added the logic and I made a test for it, does it work for you ?

Reply:
Please add such case to `tests/Fixtures/Integration/priority/multiline_long_array,array_indentation.test` too, so we can be sure that it works as expected 🙂.

Reply:
Will try to do it when I find some time this week

Reply:
I just added the test but I'm not sure what it was supposed to do.

I was expecting :
```php
// from
[$foo, $bar, $baz] = $arr;

// to
[
    $foo,
    $bar,
    $baz
] = $arr;
```
But I got :
```php
[
$foo,
$bar,
$baz
] = $arr;
```

After looking around, it seems like `ArrayIndentation` does not work with destructuring, right ?
So is this what was expected ? If not, is the test still relevant ?

Reply:
It'll work when #7405 is merged, so we need to update integration test here when it's done 🙂.

Reply:
Got you :+1: I just added the test